library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -261,5 +276,29 @@ private boolean isAnnotationNullable(AnnotationExpr annotation) {
               && this.isJspecifyNullableImportPresent)
           || annotation.getNameAsString().equalsIgnoreCase(JSPECIFY_NULLABLE_IMPORT);
     }
+
+    /**
+     * Takes a ClassOrInterfaceDeclaration instance and returns a Map of the indexes and a set of
+     * annotations for generic type parameters with Nullable upper bounds.
+     *
+     * @param cid ClassOrInterfaceDeclaration instance.
+     * @return Map of annotations for generic type parameters with Nullable upper bounds.
+     */
+    private ImmutableMap<Integer, ImmutableSet<String>> getGenericTypeParameterNullableUpperBounds(
```

Comment:
Returning a map seems overly complex.  If we are only looking for `@Nullable` upper bounds why not just return an `ImmutableSet<Integer>` containing those type variable indices with such a bound?